### PR TITLE
Added Kubectl equivalent example for Creating Resource Quota 

### LINKED ISSUE #4267: Missing Kubectl Equivalents
### Is your enhancement related to a problem? Please describe

I found that some equivalents are missing in kubernetes-examples/kubectl/equivalents as shown below:

- [x] create resource quota yaml
- [x] create namespace 
- [ ] delete namespace
- [ ] get namespace by label
- [ ] ..more

### Describe the solution you'd like

I have already started working on this and will raise a PR soon. Please assign this task to me.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Ananya2001-an**
## Description

Fix #4267

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

In ref. to issue #4267

### CODE REVIEW

**Comment by rohanKanojia** [comment]
@Ananya2001-an : Could you please address build failures?

**Comment by rohanKanojia** [comment]
@Ananya2001-an : Build failure isn't caused by failing tests. It seems to be a compilation issue:

```
Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
/home/runner/work/kubernetes-client/kubernetes-client/kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java:[33,83] cannot find symbol
  symbol:   variable myspace
```

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * It loads YAML manifest with specified name and creates it.
+ */
+
+public class CreateResourceQuotaInNamespaceYamlEquivalent{
+	public static void main(String[] args) {
+		
+		try(KubernetesClient client = new KubernetesClientBuilder().build())
+		{
+			ResourceQuota quota = client.resourceQuotas().inNamespace(myspace).createOrReplace(
+            new ResourceQuotaBuilder().withNewMetadata().withName("quota").endMetadata()
```
Comment (rohanKanojia) [inline kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java:34]: I think you're following Kubernetes docs https://kubernetes.io/docs/concepts/policy/resource-quotas/

You also need to provide spec for ResourceQuota

Comment (Ananya2001-an) [reply kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java:34]: Got it !

**Comment by Ananya2001-an** [comment]
Made requested changes

**Comment by rohanKanojia** [comment]
@Ananya2001-an : You still haven't addressed compilation failures. Build is going to fail again. Have you verified your changes build okay using `mvn clean install` as mentioned in [Contributions Guide](https://github.com/fabric8io/kubernetes-client/blob/master/CONTRIBUTING.md)?

**Comment by Ananya2001-an** [comment]
@rohanKanojia No I didn't. I am really sorry for the inconvenience. I will properly test everything now and then push commits.

**Comment by Ananya2001-an** [comment]
@rohanKanojia I have solved the compilation issue but after running tests I am still getting build failure due to some serialization test. 
The only problem with my changes was the variable myspace which I have now converted to a string. And I feel it should work now.

**Comment by rohanKanojia** [comment]
@Ananya2001-an : Could you please push your changes?

**Comment by Ananya2001-an** [comment]
@rohanKanojia yes I will

**Comment by Ananya2001-an** [comment]
@rohanKanojia Pushed changes. Kindly check.

**Comment by rohanKanojia** [comment]
Yes, I don't think you verified your changes. If you check builg logs you should be able to see them

**Comment by Ananya2001-an** [comment]
How to check the build logs after running mvn clean install?

**Comment by rohanKanojia** [comment]
I'm not able to understand your question. Are you asking how to check build job logs? 

https://github.com/fabric8io/kubernetes-client/runs/7447480859?check_suite_focus=true

**Comment by Ananya2001-an** [comment]
@rohanKanojia I have not rebased my PR till now. Do u think that's the cause for compilation error?
I ran the tests again after changing namespace to "default" but still got the same error. 
I have done the formatting of files as well by calling mvn spotless:apply which resulted in success but other tests were failing as usual.

**Comment by rohanKanojia** [comment]
I'm not sure how to make myself more clear. Build is failing because you're missing an import. It's just a minor compilation issue:
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project kubernetes-examples: Compilation failure
Error:  /home/runner/work/kubernetes-client/kubernetes-client/kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java:[35,50] cannot find symbol
Error:    symbol:   class Quantity
Error:    location: class io.fabric8.kubernetes.examples.kubectl.equivalents.CreateResourceQuotaInNamespaceYamlEquivalent
Error:  -> [Help 1]
```

**Comment by Ananya2001-an** [comment]
Okay got it. Sorry for the trouble.

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import io.fabric8.kubernetes.api.model.Quantity;
+import io.fabric8.kubernetes.api.model.ResourceQuota;
+import io.fabric8.kubernetes.api.model.ResourceQuotaBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+
+/**
+ * This sample code is Java equivalent to `kubectl create -f resources.yaml --namespace=myspace`.
+ * It loads YAML manifest with specified name and creates it.
```
Comment (rohanKanojia) [inline kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java:26]: This javadoc description doesn't align with what code is doing. It says YAML manifest gets loaded but in code we're actually creating an object inline.

Either this needs to be changed or you need to update code to use `.load()` method to load ResourceQuota provided in `resources.yaml` file

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+public class CreateResourceQuotaInNamespaceYamlEquivalent {
+  public static void main(String[] args) {
+
+    try (KubernetesClient client = new KubernetesClientBuilder().build()) {
+      ResourceQuota quota = client.resourceQuotas().inNamespace("default").createOrReplace(
```
Comment (rohanKanojia) [inline kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java:33]: `createOrReplace(..)` is deprecated, Use `resource(..).createOrReplace()` instead

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java (hunk 2 above)
Comment (rohanKanojia) [reply kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java:26]: Also you'd need to add an entry in this table for this file https://github.com/fabric8io/kubernetes-client#kubectl-java-equivalents

**Comment by rohanKanojia** [comment]
@Ananya2001-an : polite ping, are you still working on this?

**Comment by Ananya2001-an** [comment]
@rohanKanojia Hey. Sorry for the late response. I do want to continue working on this. I am actually bit occupied with some work which will take 2 more weeks to finish. But after that I will definitely make the suggested changes. I hope it's fine.

**Comment by manusa** [comment]
Hi @Ananya2001-an
This PR is becoming stale, are you still working on this?

**Comment by Ananya2001-an** [comment]
Hi @manusa 
I will do it within 3 days. Really sorry for the delay. Got caught up in some work.

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+public class CreateResourceQuotaInNamespaceYamlEquivalent {
+  public static void main(String[] args) {
+
+    try (KubernetesClient client = new KubernetesClientBuilder().build()) {
+      /*
+       * If namespace is specified in YAML client would pick it up from there,
+       * otherwise you would need to specify it in operation context like being done
+       * here.
+       */
+      client.load(CreateResourceQuotaInNamespaceYamlEquivalent.class.getResourceAsStream("/resources.yaml"))
```
Comment (sunix) [inline kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java:35]: Hello, I think we are we missing the `resources.yaml` file ?

Comment (Ananya2001-an) [reply kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/CreateResourceQuotaInNamespaceYamlEquivalent.java:35]: Hi @sunix I am not sure how to create the resources.yaml file for this example. Can you guide please?

**Review by rohanKanojia — APPROVED** [review]

**Review by sunix — APPROVED** [review]

**Review by manusa — APPROVED** [review]

**Pull request merged**